llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -1897,49 +1888,68 @@ Error MCCASBuilder::createLineSection() {
   uint64_t *OffsetPtr = &Prologue->Offset;
   uint64_t End =
       Prologue->Length + (Prologue->Format == dwarf::DWARF32 ? 4 : 8);
-  uint64_t LineTableStartOffset = Prologue->Offset;
   while (*OffsetPtr < End) {
     DWARFDataExtractor::Cursor Cursor(*OffsetPtr);
     auto Opcode = LineTableDataReader.getU8(Cursor);
+    LineTableData.push_back(Opcode);
     if (Opcode == 0) {
       // Extended Opcodes always start with a zero opcode followed by
       // a uleb128 length so unknown opcodes can be skipped.
+      uint64_t CurrOffset = Cursor.tell();
       uint64_t Len = LineTableDataReader.getULEB128(Cursor);
       if (Len == 0)
         return createStringError(inconvertibleErrorCode(),
                                  "0 Length for an extended opcode is wrong");
+      copyData(LineTableData, DebugLineStrRef, CurrOffset, Cursor);
 
       uint8_t SubOpcode = LineTableDataReader.getU8(Cursor);
+      LineTableData.push_back(SubOpcode);
       bool IsEndSequence = false;
-      auto Err = handleExtendedOpcodesForLineTable(
-          LineTableDataReader, Cursor, SubOpcode, Len, IsEndSequence);
+      bool IsRelocation = false;
+      CurrOffset = Cursor.tell();
+      auto Err = handleExtendedOpcodesForLineTable(LineTableDataReader, Cursor,
+                                                   SubOpcode, Len,
+                                                   IsEndSequence, IsRelocation);
       if (Err)
         return Err;
+      if (IsRelocation) {
+        LineTableDataReader.getRelocatedAddress(Cursor);
```

Comment:
As a casual reader I'm wondering why the result of this isn't needed. Should this be explained in a comment?

Reply:
I am just moving the cursor by the size of the relocation, which I am then copying on the next line, but I can add a comment for that too